.pre-commit-config.yaml

```diff
@@ -13,7 +13,13 @@ repos:
         name: Detect hardcoded secrets
         description: Detect hardcoded secrets using Gitleaks
         language: docker_image
-        entry: zricethezav/gitleaks:latest protect --verbose --redact --staged
+        entry: >
+          -u root:root
```

Comment:
why is `-u root:root` needed?

Reply:
a bug in gitleaks where the docker container is having permission issues(?) and cannot detect the git repo. This is the workaround found here:
- https://github.com/gitleaks/gitleaks/issues/1364#issuecomment-2035545023